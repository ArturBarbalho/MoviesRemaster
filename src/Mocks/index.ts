import { BannerProps, CardProps, bigbannerProps } from "../typesProps";
export const BiggerBannerMock:bigbannerProps = {
  bigsrc: 'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
  smallsrc:'https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg',
  genres:['Action', 'Comedy'],
  id:2,
  media:'movie',
  overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
  title:'ONYANKOPON 3 O IN√çCIO'
}
export const CardMock:CardProps = {
  id:2,
  media:'movie',
  src:'http://www.claquete.com/fotos/filmes/poster/12861_medio.jpg',
  title:'ONYANKOPON 2 O RETORNO'
}

export const BannerMock:BannerProps = {
    media: 'movie',
    relaseDate: '(1800)',
    id:1,
    title:'ONYANKOPON',
    src:'https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg',
    overview:'Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...',
    genres: ['Action', 'Comedy' ]
  }