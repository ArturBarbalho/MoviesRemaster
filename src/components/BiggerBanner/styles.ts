import styled from 'styled-components'
import media from 'styled-media-query'
export const Wrapper = styled.div``
export const Cont = styled.div`
display:flex;
max-width:2000px;
justify-content:center;
margin-top:100px;
${media.lessThan('small')`
    margin-top:0px;
`}

`
export const ButtonDiv = styled.div`
display: flex;
justify-content:center;
`