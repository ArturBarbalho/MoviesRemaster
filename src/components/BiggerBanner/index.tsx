import Button from '../Button'
import Genres from '../Genres'
import Image from '../Image'
import Overview from '../Overview'
import * as S from './styles'
import {Functions} from '../../getdata/index'
import { bigbannerProps } from '../../typesProps'

type props = {
  props:bigbannerProps
}

const BiggerBanner = ({props}:props) => (
  <S.Wrapper>
    <Image src={props.bigsrc} size='larger' />
    <S.Cont data-testid='cont'>  
        <Image size='medium' src={props.smallsrc} />  
        <div>               
        <Genres genres={props.genres} size='bigger' /> 
        <Overview size='medium' overview={props.overview}/>
        <S.ButtonDiv>
        <Button onClick={()=>Functions.Details(props.title,props.id,props.media)} color='gray'/> 
        </S.ButtonDiv>                         
        </div>                                                   
    </S.Cont>
  </S.Wrapper>
)

export default BiggerBanner