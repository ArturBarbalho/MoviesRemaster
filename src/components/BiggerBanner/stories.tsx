import { Story, Meta } from '@storybook/react/types-6-0'
import BiggerBanner from '.'
import { BiggerBannerMock } from '../../Mocks'

export default {
  title: 'BiggerBanner',
  component: BiggerBanner
} as Meta

export const Default: Story = () => <BiggerBanner props={BiggerBannerMock} />