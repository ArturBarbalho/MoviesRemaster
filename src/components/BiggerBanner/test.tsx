import { render, screen } from '@testing-library/react'

import BiggerBanner from '.'
import { BiggerBannerMock } from '../../Mocks'

describe('<BiggerBanner />', () => {
  it('should render the flex div', () => {
    const { container } = render(<BiggerBanner props={BiggerBannerMock} />)

    expect(screen.getByTestId('cont')).toHaveStyle(
      'display:flex;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })

})