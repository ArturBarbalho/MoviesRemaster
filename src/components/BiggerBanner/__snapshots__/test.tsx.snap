// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BiggerBanner /> should render the flex div 1`] = `
<div
  class="styles__Wrapper-sc-18y4bwb-0"
>
  <img
    class="styles__Wrapper-sc-y6e3xk-0 gHPgtQ"
    src="https://image.tmdb.org/t/p/original//9VIURlZvfnmFACs2m0MdUDDC3In.jpg"
  />
  <div
    class="styles__Cont-sc-18y4bwb-1 kgmLvR"
    data-testid="cont"
  >
    <img
      class="styles__Wrapper-sc-y6e3xk-0 cePpyH"
      src="https://image.tmdb.org/t/p/original//ifRFLx83Xk1DcwAS3OScgI6HmWO.jpg"
    />
    <div>
      <div
        class="styles__Wrapper-sc-6y5ihb-1 daGpDx"
      >
        <div
          class="styles__Genre-sc-6y5ihb-0 egsKaq"
        >
          Action
        </div>
        <div
          class="styles__Genre-sc-6y5ihb-0 egsKaq"
        >
          Comedy
        </div>
      </div>
      <p
        class="styles__Wrapper-sc-mhbed0-0 gpZHFY"
      >
        Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis...
      </p>
      <div
        class="styles__ButtonDiv-sc-18y4bwb-2 brWXpG"
      >
        <button
          class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
          color="gray"
        >
          See more
        </button>
      </div>
    </div>
  </div>
</div>
`;
